### PR TITLE
Make it so nanobox runs in the correct context

### LINKED ISSUE #451: Disallow running as superuser
Many new user issues are caused by using `sudo` to run various commands. We should detect when Nanobox is being run with superuser rights already available, and bail out with a warning similar to the following:

> Nanobox should not be run with full administrative privileges. It will ask for them when it needs them, but should otherwise be run with your normal capabilities. Please try again.

or some similar wording to avoid using `sudo`, etc.

### PR DESCRIPTION
**Pull request by lyondhill**
fixes #451

### CODE REVIEW

File: commands/commands.go [hunk 1]
```diff
@@ -55,6 +57,13 @@ var (
 				fileLogger, _ := lumber.NewAppendLogger(filepath.ToSlash(filepath.Join(config.GlobalDir(), "nanobox.log")))
 				lumber.SetLogger(fileLogger)
 
+			} else {
+				if !strings.Contains(ccmd.CommandPath(), "server")  && util.IsPrivileged() {
```
Comment (mbrodala) [inline commands/commands.go]: Maybe you should also check for CI mode too since CI systems often run as `root`

Comment (lyondhill) [inline commands/commands.go]: I will look into it. I know travis can allow you to run as admin but they dont run as admin without using 'sudo'

**Comment by danhunsaker** [comment]
Won't configuration commands apply to the root user when run as root? Thought the config was a per-user thing.

**Comment by lyondhill** [comment]
yah thats why i added https://github.com/nanobox-io/nanobox/pull/457/files#diff-b54b33b58b9d76939b70cd4d2f48f251R69

**Comment by danhunsaker** [comment]
I understand needing to be able to run config commands as root if the entire app will run as root, but outside of CI mode (or turning that on in the first place), that's not desired, is it?

**Comment by lyondhill** [comment]
the problem is if your admin and your trying to set cimode you wouldnt be able to unless we allow config commands as admin

**Comment by danhunsaker** [comment]
Yeah. Just saying it might be useful to be a bit more picky about *which* config values can be changed as root, since part of the problem that led to this PR in the first place was that users keep switching between `sudo` and regular invocations. They'd configure as normal, then run as root (usually for Docker access), or configure as root and run as normal. It causes a lot of "Implode as root, and start over as yourself" responses.

**Comment by lyondhill** [comment]
thats true. I could change it to say only allow `set ci-mode`

**Pull request merged**